core/variant/variant_parser.cpp

```diff
@@ -1964,11 +1972,17 @@ Error VariantWriter::write(const Variant &p_variant, StoreStringFunc p_store_str
 			p_store_string_func(p_store_string_ud, itos(p_variant.operator int64_t()));
 		} break;
 		case Variant::FLOAT: {
-			String s = rtos_fix(p_variant.operator double(), p_compat);
-			if (s != "inf" && s != "-inf" && s != "nan") {
-				if (!s.contains_char('.') && !s.contains_char('e') && !s.contains_char('E')) {
-					s += ".0";
-				}
+			const double value = p_variant.operator double();
+			String s;
+			// Hack to avoid garbage digits when the underlying float is 32-bit.
```

Comment:
So I found that this "Hack" is also necessary on all the Vectors, AABBs, etc. when compiling with `precision=double`. That or the current grisu2 implementation isn't fully stable for double precision.